sstables/sstable_directory.cc

```diff
@@ -62,8 +62,8 @@ sstable_directory::sstable_directory(sstables_manager& manager,
     , _unshared_remote_sstables(smp::count)
 {}
 
-void sstable_directory::components_lister::handle(sstables::entry_descriptor desc, fs::path filename) {
-    if ((generation_value(desc.generation) % smp::count) != this_shard_id()) {
+void sstable_directory::components_lister::handle(sstables::entry_descriptor desc, fs::path filename, process_flags flags) {
+    if (flags.check_shard_ownership && (generation_value(desc.generation) % smp::count) != this_shard_id()) {
```

Comment:
When adding support for uuid-based generation this method of desc.gen % smp::count will no longer work.
I presume we would want to pass a function from the caller of the lister to filter sstable opening per shard based on whatever hints are available, including l, but not limited to the numeric generation.

Myabe it's a good opportunity to introduce such predicate, that in the tool case would simply return `true` to all sstables (could be the default function if not overriden by the caller).

Cc @raphaelsc @tchaikov

Reply:
I would prefer not to do this in this PR. Changes like this are best done in the context of the code which actually requires them. Otherwise there is a good chance I will not get it right, and it will have to be amended anyway.

Reply:
thank you, Benny for bringing this up! but probably we can focus on what this PR is aiming at in this changeset. and try to generalize various interfaces for uuid-based generation in more dedicated PR(s).

Reply:
By the way, the hint is irrelevant for UUID work after my recent series. It works regardless of the generation type.

Reply:
Problematic patch dropped in v3.